### PR TITLE
[media] changing FFMpeg MACROs to function calls.

### PR DESCRIPTION
**Pull request by sideb0ard**
b/184697040

### CODE REVIEW

**Review by kaidokert — APPROVED** [review]
This looks good. Could use a bit more extensive description in PR / commit message, e.g. "Resolves version differences at library load-time, rather than compile time"

File: starboard/shared/ffmpeg/ffmpeg_dynamic_load_dispatch_impl.cc [hunk 1]
```diff
@@ -247,39 +247,55 @@ bool FFMPEGDispatchImpl::OpenLibraries() {
...
+  errstr = dlerror();                                                   \
+  if (errstr != NULL) {                                                 \
+    SB_LOG(INFO) << "Load Symbols ran into error:" << errstr;           \
+  }
 
   // Load symbols from the avutil shared library.
   INITSYMBOL(avutil_, avutil_version);
   SB_DCHECK(ffmpeg_->avutil_version);
+  SB_LOG(INFO) << "Opened avutil  - version is:" << ffmpeg_->avutil_version()
+               << std::endl;
```
Comment (kaidokert) [inline starboard/shared/ffmpeg/ffmpeg_dynamic_load_dispatch_impl.cc:271]: Should we add a DCHECK / CHECK or at least a warning around maximum/minimum known and supported versions ? I'm guessing there might be a an upcoming version that will have changed its API again and our current code will try to use the API in a certain way that isn't supported. When it inevitably crashes this log message might save a couple hours for someone troubleshooting.

**Comment by sideb0ard** [comment]
> This looks good. Could use a bit more extensive description in PR / commit message, e.g. "Resolves version differences at library load-time, rather than compile time"

I have a doc which further describes the troubleshooting steps and findings at go/cobalt-ffmpeg-forge-bug

We were using one definition of LIBAVUTIL_VERSION_INT to gate certain includes, however, that value is different or not defined, depending on which file includes the header. This change here, rather than relying on the correct ordering of includes, resolves the function calls at runtime based upon the actual version of the library found on the system.

**Review by jellefoks — APPROVED** [review]

File: starboard/shared/ffmpeg/ffmpeg_dynamic_load_dispatch_impl.cc (hunk 1 above)
Comment (sideb0ard) [reply starboard/shared/ffmpeg/ffmpeg_dynamic_load_dispatch_impl.cc:271]: I don't think we should see breaking API changes. From what I can see its usually more about when a new API was added. We use quite a small subset of the FFMPEG function calls.

**Pull request merged**